backend/domain/domainmodels.js

```diff
@@ -0,0 +1,134 @@
+
+
+
+
+function Room (){
+
+        this.roomName = this.setRoomName(); //@@@@@@@ -some function to initialise a roomID
+
+        this.gamestate = GameStateEnum["Day Time"];
+
+        this.maxPlayerCount = 6;
+
+        this.players = new Array(maxPlayerCount);
+
+        this.roundNumber = 0;
+
+        this.getRoomName = function () {
+            return this.roomName;
+        };
+
+        this.getGamestate = function () {
+            return this.gamestate;
+        };
+
+        this.setGamestate = function (gamestate) {
+            this.gamestate = gamestate;
+        };
+
+
+        this.getPlayers = function () {
+            return this.players;
+        };
+
+
+        this.addPlayer = function (player) {
+            if (player !== null &&
+                this.getPlayers.length <= maxPlayerCount) {
+                this.players.push(player);
+            }
+        };
+
+
+        this.getRoundNumber = function () {
+            return this.roundNumber;
+        };
+
+        this.incrementRoundNumber = function () {
+            this.roundNumber++;
+        };
+
+        this.setRoomName = function () {
+            return 0; //@@@@@@@@@@@@@//@@@@@@@ -some function to initialise a roomID
+        };
+
+
+}
+
+
+
+
+function Player(socketID, nickname, role){
+    this.socketID = socketID;
+    this.nickname = nickname
+    this.role = role
+    this.isAlive = true
+
+
+    this.getName = function(){
+        return this.name
+    } 
+
+    this.getNickname = function(){
+        return this.nickname
+    } 
+
+    this.getRole = function(){
+        return this.role
+    } 
+
+    this.getIsAlive = function(){
+        return this.isAlive
+    } 
+
+    this.setIsAlive = function(){
+        this.isAlive = !this.isAlive
+    } 
+
+} 
+
+
+
+
+function MafiaGame(){
+    const dict = {}
+    this.gameCount = 0
+    //Map(roomID, Room)
+
+    this.newGame = function(){ 
+        incrementGameCoutner()
+        newRoom = new Room()
+        gameID = newRoom.getRoomName()
+        //https://stackoverflow.com/questions/7196212/how-to-create-dictionary-and-add-key-value-pairs-dynamically
+        dict[gameID] = newRoom
+
+    }
+
+    this.closeGame = function(gameID){ 
+        decrementGameCoutner()
+        //https://stackoverflow.com/questions/346021/how-do-i-remove-objects-from-a-javascript-associative-array
+        delete dict[gameID]  
+        
+    }
+
+
+    this.incrementGameCoutner = function(){ 
+        this.gameCount++
+    }
+    
+    this.decrementGameCoutner = function(){ 
+        this.gameCount--
+    }
+
+}
+
+
+//wtf is this
+const RoleEnum = {"Civilian":1, "Mafia":2, "Medic":3, "Detective":4}
```

Comment:
Should we have a space between the colon and the value for properties? For example, Civilian: 1, Mafia: 2, and so on

Reply:
This is the notation of Artur Czajka from the post explaining enumerated types in JavaScript:
https://stackoverflow.com/questions/287903/what-is-the-preferred-syntax-for-defining-enums-in-javascript

It can be changed if requested though.